api/pacman-api-compliance/src/main/java/com/tmobile/pacman/api/compliance/repository/FilterRepositoryImpl.java

```diff
@@ -297,15 +298,37 @@ public AssetCountDTO[] getValueListforTag(String assetGroup, String tag, String
      * @see com.tmobile.pacman.api.compliance.repository.FilterRepository#
      * getRegionsFromES(java.lang.String)
      */
-    public Map<String, Long> getNotificationTypesFromES()
+    public Map<String, Long> getNotificationTypesFromES(Map<String, List<String>> filter)
             throws DataException {
         Map<String, Object> mustFilter = new HashMap<>();
         Map<String, Object> mustNotFilter = new HashMap<>();
+        Map<String, Object> mustTermsFilter = new HashMap<>();
+
+        List<String> eventSource = filter.get(NOTIFICATION_SOURCE_NAME);
+        List<String> eventName = filter.get(NOTIFICATION_EVENT_NAME);
+
+        if(eventSource!=null && !eventSource.isEmpty()) {
+            mustFilter.put("eventSourceName.keyword", eventSource);
+        }
+
+        if(eventName!=null && !eventName.isEmpty()) {
+            mustFilter.put("eventName.keyword", eventName);
+        }
+
+        if(!mustFilter.isEmpty()) {
+            mustTermsFilter.putAll(mustFilter);
+        }
+
+        Map<String ,Object> dateFilterMap=getDateFilter(filter);
+
+        if(dateFilterMap!=null  && !dateFilterMap.isEmpty()){
+            mustFilter.put("range",dateFilterMap);
+        }

```

Comment:
Just highlight this area and press Ctrl-Alt-L (InteliJ), it will do reformat for you

Reply:
done